packages/bonito-ui/src/components/form/action-form.tsx

```diff
@@ -146,6 +151,56 @@ export const ActionForm = <V extends FormValues>(
         return <></>;
     }
 
+    // Unified form submission logic
+    const handleSubmit = async () => {
+        setSubmitting(true);
+        try {
+            // Store current form values before execution
+            setLastTypedValues(action.form.values);
```

Comment:
`lastTypedValues` doesn't have to be a react state. Can be achieved by using a local variable
```suggestion
            const lastTypedValues = action.form.values
            ...
            action.form.setValues(lastTypedValues)
```